File: php/src/APIError.php
<?php

namespace CCPayment;

use Exception;

class APIError extends Exception
{
    private int $code;

    public function __construct(int $code, string $message)
    {
        parent::__construct("API Error {$code}: {$message}");
        $this->code = $code;
    }

    public function getErrorCode(): int
    {
        return $this->code;
    }
}

File: java/src/main/java/com/ccpayment/APIError.java
package com.ccpayment;

public class APIError extends Exception {
    private final int code;
    private final String message;

    public APIError(int code, String message) {
        super("API Error " + code + ": " + message);
        this.code = code;
        this.message = message;
    }

    public int getCode() {
        return code;
    }

    @Override
    public String getMessage() {
        return message;
    }
}
